pkgs/development/compilers/dotnet/sdk/stage0.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenv
+, callPackage
+, lib
+, writeShellScript
+, pkgsBuildHost
+, mkNugetDeps
+, cacert
+, nuget-to-nix
+, dotnetCorePackages
+
+, releaseManifest
+, hash
+, prepDepsFile
+, buildDepsFile
+, bootstrapSdk
+}:
+
+let
+  mkPackages = callPackage ./packages.nix;
+  mkVMR = callPackage ./vmr.nix;
+
+  dotnetSdk = pkgsBuildHost.callPackage bootstrapSdk {};
+
+  patchNupkgs = pkgsBuildHost.callPackage ./patch-nupkgs.nix {};
+
+  vmr = (mkVMR {
+    inherit releaseManifest hash dotnetSdk;
+  }).overrideAttrs (old: rec {
+    prepDeps = mkNugetDeps {
+      name = "${old.pname}-prep";
+      sourceFile = prepDepsFile;
+    };
+
+    buildDeps = mkNugetDeps {
+      name = "${old.pname}-build";
+      sourceFile = buildDepsFile;
+    };
+
+    patches = old.patches or [] ++ [
+      ./use-nuget-deps.patch
+      ./patch-restored-packages.patch
+    ];
+
+    nativeBuildInputs = old.nativeBuildInputs ++ [
+      patchNupkgs
+    ];
+
+    postConfigure = old.postConfigure or "" + ''
+      [[ ! -v buildDeps ]] || ln -sf "$buildDeps"/* prereqs/packages/prebuilt/
+    '';
+
+    passthru = old.passthru or {} // { fetch-deps =
+      let
+        inherit (vmr) targetRid;
+        otherRids =
+          lib.remove targetRid (
+            map (system: dotnetCorePackages.systemToDotnetRid system)
+              vmr.meta.platforms);
+
+        pkg = vmr.overrideAttrs (old: {
+          patches = old.patches or [] ++ [
+            ./allow-missing-pdbs.patch
+            ./record-downloaded-packages.patch
+          ];
+          nativeBuildInputs = old.nativeBuildInputs ++ [
+            cacert
+            (nuget-to-nix.override { dotnet-sdk = dotnetSdk; })
+          ];
+          buildFlags = [ "--online" ] ++ old.buildFlags;
+          prepDeps = null;
+          buildDeps = null;
+        });
+
+        drv = builtins.unsafeDiscardOutputDependency pkg.drvPath;
+      in
+        writeShellScript "fetch-dotnet-sdk-deps" ''
+          set -euo pipefail
+          export PATH="${with pkgsBuildHost; lib.makeBinPath [ nix ]}":$PATH
+
+          # this needs > 16G
+          export TMPDIR=/var/tmp
```

Comment:
That directory seems to be occupied by systemd things and I am not sure if we can just dump 16GB there.

Reply:
The common fetch-deps uses:

```
        if [[ ''${TMPDIR:-} == /run/user/* ]]; then
           # /run/user is usually a tmpfs in RAM, which may be too small
           # to store all downloaded dotnet packages
           unset TMPDIR
        fi
```

I find that very hacky too.  I think I'd prefer if `fetch-deps` users just had to set TMPDIR to something large.